### PR TITLE
Change deprecated fields

### PR DESCRIPTION
**Pull request by lukidzi**
Changed few deprecated fields:
- `config` to `typed_config`
- `hosts` in clusters is deprecated changed to `load_assignment`
-  changed default Envoy version to `v1.14.3`
- changed deprecated `ORIGINAL_DST_LB` to `CLUSTER_PROVIDED`

### CODE REVIEW

File: envoy-control-tests/src/main/resources/envoy/echo3_envoy_auth_config.yaml [hunk 1]
```diff
@@ -0,0 +1,96 @@
+admin:
```
Comment (pzmi) [inline envoy-control-tests/src/main/resources/envoy/echo3_envoy_auth_config.yaml:1]: Is this a new config file? Where is it used?

Comment (lukidzi) [reply envoy-control-tests/src/main/resources/envoy/echo3_envoy_auth_config.yaml]: Didn't noticed them during merge. Removed

**Review by pzmi — DISMISSED** [review]

**Review by slonka — DISMISSED** [review]

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/EnvoyControlTestConfiguration.kt [hunk 2]
```diff
@@ -83,8 +83,7 @@ abstract class EnvoyControlTestConfiguration : BaseEnvoyTest() {
         var envoyControls: Int = 1
         var envoys: Int = 1
 
-        // We use envoy version from master. This is 1.14.0-dev.
-        const val defaultEnvoyImage = "envoyproxy/envoy-alpine-dev:5b1723ff54b1a51e104c514ee6363234aaa44366"
+        const val defaultEnvoyImage = "envoyproxy/envoy-alpine:v1.14.3"
```
Comment (MarcinFalkowski) [inline envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/EnvoyControlTestConfiguration.kt:86]: I know that envoy version is hardcoded in at least one more place in this repo :)

Comment (MarcinFalkowski) [reply envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/EnvoyControlTestConfiguration.kt:86]: https://github.com/allegro/envoy-control/blob/master/tools/envoy/Dockerfile

Comment (slonka) [reply envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/EnvoyControlTestConfiguration.kt:86]: @lukidzi can you resolve this?

**Review by pzmi — APPROVED** [review]

**Review by MarcinFalkowski — APPROVED** [review]

**Pull request merged**